components/threading/global_threads/thread_list/virtualized_thread_list.tsx

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {memo, useRef, useCallback, useMemo} from 'react';
+import AutoSizer from 'react-virtualized-auto-sizer';
+import {DynamicSizeList} from 'dynamic-virtualized-list';
+
+import {$ID} from 'mattermost-redux/types/utilities';
+import {UserThread} from 'mattermost-redux/types/threads';
+
+import ThreadItem from '../thread_item';
+
+type Props = {
+    ids: Array<$ID<UserThread>>;
+    selectedThreadId?: $ID<UserThread>;
+};
+
+const ITEMS_ON_SCREEN = 8;
+const OVERSCAN_COUNT_BACKWARD = 4;
+const OVERSCAN_COUNT_FORWARD = 4;
+
+const virtListStyles = {
+    maxHeight: '100%',
+    position: 'absolute',
+    top: '0',
+};
+
+function VirtualizedThreadList({
+    ids,
+    selectedThreadId,
+}: Props) {
+    const innerRef = useRef();
+
+    const renderRow = ({itemId}: {itemId: $ID<UserThread>}) => {
+        return (
+            <ThreadItem
+                key={itemId}
+                threadId={itemId}
+                isSelected={selectedThreadId === itemId}
+            />
+        );
+    };
+
+    const canLoadMorePosts = useCallback(() => {}, []);
+
+    // dynamic-virtualized-list renders items in reverse order
+    // so we need to reverse the ids before feeding the list
+    // so that we get the correct order afterwards.
+    //
+    // TODO: have dynamic-virtualized-list to maybe accept a direction prop
```

Comment:
nitpick: this should be an issue in the [repo](https://github.com/mattermost/dynamic-virtualized-list) instead of a todo here

Reply:
Note that the repo was created originally by Sudheer as a fork of react-window, and since he left, I don't think anyone else has been maintaining it.

Reply:
> Edit: Also, that's funny that it works that way since I'm pretty sure the reason it does that is because we send the posts to the client sorted backwards (newest to oldest) which honestly has caused a lot of confusion in the post storage code in the past. I wonder if we should just store posts oldest to newest and then update the list component to never reverse the list.

@hmhealey, it seems that it doesn't work :( I cannot find a way to add items at the top... 
items added at the bottom (0 index) are getting rendered, but items added at the top (last index) don't...

I'd like to add back `react-window` for this case.